File: solutions/kotlin/resistor-color/2/src/main/kotlin/ResistorColor.kt
enum class Color {
    BLACK, BROWN, RED, ORANGE, YELLOW,
    GREEN, BLUE, VIOLET, GREY, WHITE,
}

object ResistorColor {
    fun colorCode(input: String): Int =
        Color.valueOf(input.uppercase()).ordinal

    fun colors(): List<String> =
        Color.entries.map{ entry ->
            entry.name.lowercase()
        }.toList()
}

File: solutions/go/resistor-color/2/resistor_color.go
package resistorcolor

import "slices"

var colors = [10]string {
    "black", "brown", "red", "orange", "yellow",
    "green", "blue", "violet", "grey", "white",
}

func Colors() (s []string) {
    return slices.Clone(colors[:])
}

func ColorCode(color string) int {
	return slices.Index(Colors(), color)
}